src/components/AdvancedSearchBar/AdvancedSearchBar.vue

```diff
@@ -0,0 +1,181 @@
+          @click="clearSearch"
+        >
+          <XmarkLarge />
+        </button>
+      </template>
+    </TextInput>
+    <div
+      v-if="visible || searchTerm.value"
+      class="bg-white shadow overflow-y-auto min-w-full absolute"
+      role="results"
+    >
+      <!-- If search is still running do not perform any action and show a searching bar -->
+      <template v-if="searchTerm && searching">
+        {{ t('search.loading') }}
+      </template>
+      <!-- If search is done and seach has results, show the results -->
+      <template v-else-if="!searching && searchResults">
+        <div class="search-body m-2 border-b">
+          <div v-for="itemGroup in searchResults" :key="itemGroup">
+            <LobTable v-if="itemGroup" class="min-w-full mb-4" space="sm">
+              <TableHeader>
+                <slot name="header" :result="itemGroup" />
+              </TableHeader>
+              <TableBody>
+                <TableRow
+                  v-for="item in itemGroup?.items"
+                  :key="item"
+                  class="text-gray-500 hover:text-primary-700 cursor-pointer"
+                  @click="hide"
+                >
+                  <slot name="body" :result="item" />
+                </TableRow>
+              </TableBody>
+            </LobTable>
+          </div>
+        </div>
+        <!-- If search has results and user wants to show a footer, the total number of results -->
+        <div
+          v-if="searchTerm && footer"
+          class="search-footer m-2 flow-root text-sm text-gray-700"
+        >
+          <div class="float-left">
+            {{ totalResults }} {{ t('search.matchingResults') }}
+          </div>
+          <div class="float-right flex">
+            <LobLink
+              :to="link"
+              :underline="false"
+              class="text-sm hover:text-primary-900"
+              @click="hide"
+            >
+              {{ t('search.seeAllResults') }}
+              <Icon class="inline" :icon="IconName.NEXT" />
+            </LobLink>
+          </div>
+        </div>
+      </template>
+      <!-- If no results are found show a no results message -->
+      <template v-else>
+        {{ t('search.noResults') }}
+      </template>
+    </div>
+  </div>
+</template>
+<script setup lang="ts">
+import TextInput from '../TextInput/TextInput';
+import LobTable from '../Table/Table';
+import TableHeader from '../Table/TableHeader';
+import TableBody from '../Table/TableBody';
+import TableRow from '../Table/TableRow';
+import LobLink from '../Link/Link.vue';
+import MagnifyingGlass from '../Icons/MagnifyingGlass';
+import XmarkLarge from '../Icons/XmarkLarge';
+import { ref, watch, computed, onMounted, onUnmounted } from 'vue';
+import { Icon, IconName } from '../Icon';
+
+const searchTerm = ref('');
+const searchResults = ref([]);
+const searching = ref(false);
+const visible = ref(false);
+const timeout = ref(null);
+const searchBar = ref(null);
```

Comment:
Some of these would probably benefit from being models, the `searchTerm` or `visible` would be a good use of this as I could see scenarios where we want to pre-populate that input/ manage when the results are shown

You would define it with something like `const search = defineModel<string>('search', { default: '' });` or in the case of the search term I would probably do `const modelValue = defineModel<string>({ default: '' });` and then when you use this component you can control this value with `v-model="..."`

Reply:
I understand a defineModel could be helpful in cases where we need to emit events. But none of these values emit any events to the parent module? I tried to build them in a way that they are no different from the current search component, such that it's easier to replace in the dashboard repo as well without having to build a new Search Component.

Reply:
Sounds good!